js/exporter/jspdf/v3/rows_splitting.js

```diff
@@ -29,6 +30,42 @@ function splitRectsHorizontalByPages(rects, topLeft, maxBottomRight) {
             }
         });
 
+        const currentPageCanBeSplitRects = rectsToSplit.filter(rect => {
+            // Check cells that have 'rect.x' less than 'currentPageMaxRectRight'
+            const currentRectLeft = round(rect.x);
+            const currentRectRight = round(rect.x + rect.w);
+            if(currentRectLeft < currentPageMaxRectRight && currentPageMaxRectRight < currentRectRight) {
+                return true;
+            }
+        });
+
+        currentPageCanBeSplitRects.forEach(rect => {
+            // Split merged rects that can be split and put to neccessary array
+            const leftRect = Object.assign({}, {
+                x: rect.x,
+                y: rect.y,
+                w: currentPageMaxRectRight - rect.x,
+                h: rect.h
+            }, { sourceCellInfo: rect.sourceCellInfo });
+
+            currentPageRects.push(leftRect);
+
+            const rightRect = Object.assign({}, {
+                x: currentPageMaxRectRight,
+                y: rect.y,
+                w: rect.w - (currentPageMaxRectRight - rect.x),
+                h: rect.h
+            }, { sourceCellInfo: Object.assign({}, rect.sourceCellInfo) });
+            rightRect.sourceCellInfo.text = '';
+
+            rectsToSplit.push(rightRect);
+
+            const index = rectsToSplit.indexOf(rect);
+            if(index !== -1) {
+                rectsToSplit.splice(index, 1);
```

Comment:
It seems there are smool mistake in this line
rect.x = (currentPageMaxRectRight !== undefined) ? (rect.x - currentPageMaxRectRight + topLeft.x + margin.left) : rect.x;



It doesn't took in account margin value.

It's necessary to rewrite in such way
```js
rect.x = (currentPageMaxRectRight !== undefined) ? (rect.x - currentPageMaxRectRight + topLeft.x + margin.left) : rect.x;
```

Reply:
Yes, we missed the case with margins.
I have added the tests for this case. And used this property in calculation.
Done.